File: Array/concatenationOfArray.kt
class Solution {
    fun getConcatenation(nums: IntArray): IntArray {
        return  nums+nums
    }

    fun getConcatenationV2(nums: IntArray): IntArray {
        val size = nums.size
        val result= IntArray(2*size)
        for(i in 0 until size){
            result[i] = nums[i]
            result[i+size] = nums[i]
        }

        return result
    }
}



fun main(){
   val output = Solution().getConcatenationV2(intArrayOf(1,2,1)).toList()
    println(output)
}

File: Array/concatenationOfArray.js
const getConcatenation = (array) => {
    return [...array, ...array];
}

const getConcatenationV2 = (array) => {
    return array.concat(array);
}

const getConcatenationV3 = (array) => {
    let result = [];
    const arraySize = array.length
    for (let i = 0; i < arraySize; i++) {
        result[i] = array[i]
        result[i + arraySize] = array[i]
    }
    return result
}


console.log(getConcatenation([1, 2, 3]));
console.log(getConcatenationV2([1, 2, 3]));
console.log(getConcatenationV3([1, 2, 3]));
